de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java

```diff
@@ -1039,13 +1040,12 @@ public void changeColor(int colorID) {
     }
 
     @Override
-    public void enableQueuing(IProject project) {
-        activityQueuer.enableQueuing(project);
+    public void registerQueuingHandler(IActivityQueuer activityQueuer) {
+        this.activityQueuer = activityQueuer;
```

Comment:
Check if activityQueuer is null?

I am not sure how to handle this in general. Our coderules don't specify if we generally expect the caller or the called methods to do the null checks. I generally prefer doing it in the called method, but without annotations for these kinds of checks, it adds a lot of "unnecessary" bulk to the code. ("Unnecessary" in the sens that it is not important for the reader as it usually just throws an IllegalArgumentException.)

@fzieris Any input on how to handle null checks for passed arguments in general?

Reply:
Yes, Checking on called side, especially public methods is a good idea. How about adding a short assert statement for this?

Reply:
I don't know. I generally don't work with assert statements, especially since they are not always enabled. (The tests for the Saros core were always run on our build server without asserts enabled - which lead to nobody noticing the problem described in #98 - so I would assume that normal builds are run with assertions disabled as well.

Reply:
IRC you can have multiple project negotiations for a user concurrently.

See https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/ProjectNegotiationObservable.java